machine/v1/types_controlplanemachineset.go

```diff
@@ -0,0 +1,301 @@
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp",description="ControlPlaneMachineSet age"
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+type ControlPlaneMachineSet struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ControlPlaneMachineSetSpec   `json:"spec,omitempty"`
+	Status ControlPlaneMachineSetStatus `json:"status,omitempty"`
+}
+
+// ControlPlaneMachineSet represents the configuration of the ControlPlaneMachineSet.
+type ControlPlaneMachineSetSpec struct {
+	// Replicas defines how many Control Plane Machines should be
+	// created by this ControlPlaneMachineSet.
+	// This field is immutable and cannot be changed after cluster
+	// installation.
+	// The ControlPlaneMachineSet only operates with 3 or 5 node control planes,
+	// 3 and 5 are the only valid values for this field.
+	// +kubebuilder:validation:Enum:=3;5
+	// +kubebuilder:default:=3
+	// +kubebuilder:validation:Required
+	Replicas *int32 `json:"replicas"`
+
+	// Strategy defines how the ControlPlaneMachineSet will update
+	// Machines when it detects a change to the ProviderSpec.
+	// +kubebuilder:default:={type: RollingUpdate}
+	// +optional
+	Strategy ControlPlaneMachineSetStrategy `json:"strategy,omitempty"`
+
+	// Label selector for Machines. Existing Machines selected by this
+	// selector will be the ones affected by this ControlPlaneMachineSet.
+	// It must match the template's labels.
+	// This field is considered immutable after creation of the resource.
+	// +kubebuilder:validation:Required
+	Selector metav1.LabelSelector `json:"selector"`
+
+	// Template describes the Control Plane Machines that will be created
+	// by this ControlPlaneMachineSet.
+	// +kubebuilder:validation:Required
+	Template ControlPlaneMachineSetTemplate `json:"template"`
+}
+
+// ControlPlaneMachineSetTemplate is a template used by the ControlPlaneMachineSet
+// to create the Machines that it will manage in the future.
+// +union
+// + ---
+// + This struct is a discriminated union which allows users to select the type of Machine
+// + that the ControlPlaneMachineSet should create and manage.
+// + For now, the only supported type is the OpenShift Machine API Machine, but in the future
+// + we plan to expand this to allow other Machine types such as Cluster API Machines or a
+// + future version of the Machine API Machine.
+type ControlPlaneMachineSetTemplate struct {
+	// MachineType determines the type of Machines that should be managed by the ControlPlaneMachineSet.
+	// Currently, the only valid value is machine.v1beta1.machine.openshift.io.
+	// +unionDiscriminator
+	// +kubebuilder:default:="machines.v1beta1.machine.openshift.io"
+	// +optional
```

Comment:
I see no benefit to optional, let's go required.

Reply:
Given there's only one valid value, I thought making it optional and defaulting it to that one valid value would be beneficial to users.

Is removing a default later in the cycle considered a breaking change? Eg once we add a separate machine type to this enum?